lib/elixir/lib/gen_server.ex

```diff
@@ -612,18 +612,29 @@ defmodule GenServer do
   `reason` is exit reason and `state` is the current state of the `GenServer`.
   The return value is ignored.
 
-  `c:terminate/2` is called if the `GenServer` traps exits (using `Process.flag/2`)
-  *and* the parent process sends an exit signal, or a callback (except `c:init/1`)
-  does one of the following:
+  `c:terminate/2` is useful for cleanup that requires access to the
+  `GenServer`'s state. However, it is **not guaranteed** that `c:terminate/2`
+  is called when a `GenServer` exits. Therefore, important clean-up should be
```

Comment:
```suggestion
  is called when a `GenServer` exits. Therefore, important cleanup should be
```

Thanks for the clarification! Perhaps also take this opportunity to make the language here match the rest of the docs.